packages/ra-ui-materialui/src/input/NumberInput.tsx

```diff
@@ -46,6 +47,7 @@ export const NumberInput = ({
         isRequired,
     } = useInput({
         defaultValue,
+        onBlur,
```

Comment:
Not really sure why this solves it, but it does

Reply:
It changes what you pass to the MUI `TextField` (via `...sanitizeInputRestProps(rest)`)